### PR TITLE
Added cvode functionality to SUNDIALS integrator

### PR DESCRIPTION
**Pull request by ndeak**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Extern/SUNDIALS/AMReX_SundialsIntegrator.H [hunk 1]
```diff
@@ -146,14 +155,15 @@ private:
...
         arkode_mem = nullptr;     /* empty ARKode memory structure */
+        cvode_mem = nullptr;      /* empty CVODE memory structure  */
         NLSf = nullptr;           /* empty nonlinear solver object */
         LSf = nullptr;            /* empty linear solver object    */
         inner_mem = nullptr;      /* empty ARKode memory structure */
         mristep_mem = nullptr;    /* empty ARKode memory structure */
 
         // Arbitrary tolerances
-        reltol          = 1e-4;
-        abstol          = 1e-4;
```
Comment (WeiqunZhang) [inline Src/Extern/SUNDIALS/AMReX_SundialsIntegrator.H]: Other codes might rely on the default values here, even though the comment says it's arbitrary.

**Comment by WeiqunZhang** [comment]
@dwillcox I see things like below that look suspicious.
```
 nv_many_arr = new N_Vector[NVar];
...
N_VDestroyVectorArray(nv_many_arr, NVar);
```

Where is `delete [] nv_many_arr` being called? (It cannot be inside `N_VDestroyVectorArray`, right? Because how could SUNDIALS knows to use `delete[]`, not `free` and why would SUNDIALS be responsible for deleting memory allocated in users' code?)

**Comment by WeiqunZhang** [comment]
https://github.com/LLNL/sundials/blob/1ea097bb3bce207335ac35f0b5e78df5d71c6409/src/sundials/sundials_nvector.c#L1024

Looks like `N_VDestroyVectorArray` does free the memory with `free`. In that case, we should use `malloc`, not `new` to allocate `nv_many_arr`.

**Comment by asalmgren** [comment]
@ndeak -- what is the status of this PR?   @gardner48 is happy to help on the SUNDIALS side.

**Comment by ndeak** [comment]
Last commit failed due to whitespace issue that should be resolved now, code is ready for further review if needed.

**Comment by asalmgren** [comment]
@ndeak -- have you addressed @WeiqunZhang's questions/comments above?

**Comment by WeiqunZhang** [comment]
> @ndeak -- have you addressed @WeiqunZhang's questions/comments above?

The comment has been addressed. Could either @jmsexton03 or @dwillcox review this PR?

**Review by jmsexton03 — APPROVED** [review]

**Pull request merged**